### ISSUE TITLE
RCS1046 is reported in class implementing interface

### ISSUE THREAD

**Issue by hankovich** [dotnet/roslynator#1118]
**Product and Version Used**: latest

**Actual Behavior**:
```csharp
interface IStupidInterface // from nuget
{
    Task Foo();
}

public sealed class MyAwesomeImplementation : IStupidInterface // in my solution
{
    public Task Foo() => Task.CompletedTask; // RCS1046 Asynchronous method name should end with 'Async' is reported here
}
```

**Expected Behavior**:

Methods overriding methods from the base class either methods that implement interface should be ignored even if their name does not finish with `Async`
